apps/walletd/lib/signSeedV2.ts

```diff
@@ -49,28 +40,68 @@ export function signTransactionSeedV2({
 
   const { sigHash } = sigHashResult
 
-  const addSigScResult = addSignaturesSiacoinV2({
-    mnemonic,
-    transaction,
-    addresses,
-    siacoinOutputs,
-    sigHash,
-  })
+  for (const input of transaction.siacoinInputs ?? []) {
+    // Find the index of the address in the list of addresses.
+    const indexResponse = getAddressKeyIndex({
+        address: input.parent.siacoinOutput.address,
+        addresses,
+      }
+    );
+    if ('error' in indexResponse) {
+      return { error: indexResponse.error }
```

Comment:
Is there a reason all errors are done this way instead of `throw`ing? Feels very Go-esque.

Reply:
I often find it more convenient, maybe not here but when the error(s) are used in later control flow - in this area it is consistent with how the wasm sdk has to return errors over the bridge.